### PR TITLE
Payment HSM ARM template

### PR DESCRIPTION
**Pull request by msmbaldwin**
# PR Checklist

Check these items before submitting a PR... 

[Contribution Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md)

[Best Practice Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md)

- [ ] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

## Changelog

*
*
*

### CODE REVIEW

**Comment by azure-quickstarts** [comment]
@msmbaldwin - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

**Comment by azure-quickstarts** [comment]
@msmbaldwin - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

**Comment by azure-quickstarts** [comment]
@msmbaldwin - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

**Comment by azure-quickstarts** [comment]
@msmbaldwin - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

**Comment by azure-quickstarts** [comment]
@msmbaldwin - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

**Comment by azure-quickstarts** [comment]
@msmbaldwin - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

**Comment by azure-quickstarts** [comment]
@msmbaldwin - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

**Comment by azure-quickstarts** [comment]
@msmbaldwin - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

**Comment by azure-quickstarts** [comment]
@msmbaldwin - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

**Comment by azure-quickstarts** [comment]
@msmbaldwin - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

File: quickstarts/microsoft.hardwaresecuritymodules/payment-hsm-create/azuredeploy.parameters.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentParameters.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
```
Comment (bmoore-msft) [inline quickstarts/microsoft.hardwaresecuritymodules/payment-hsm-create/azuredeploy.parameters.json:4]: for any param that does not need a unique value use a defaultValue in the template to make it easier for customers to deploy

File: quickstarts/microsoft.hardwaresecuritymodules/payment-hsm-create/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+              ],
+              "privateEndpointNetworkPolicies": "Enabled",
+              "privateLinkServiceNetworkPolicies": "Enabled"
+            }
+          }
+        ],
+        "enableDdosProtection": false
+      }
+    },
+    {
```
Comment (bmoore-msft) [inline quickstarts/microsoft.hardwaresecuritymodules/payment-hsm-create/azuredeploy.json:119]: why is this defined twice in the template?

File: quickstarts/microsoft.hardwaresecuritymodules/payment-hsm-create/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+      "defaultValue": "[resourceGroup().location]",
+      "metadata": {
+        "description": "Location for all resources."
+      }
+    },
+    "skuName": {
+      "type": "string",
+      "defaultValue": "payShield10K_LMK1_CPS60",
+      "metadata": {
+        "description": "PayShield SKU name. It must be one of the following: payShield10K_LMK1_CPS60, payShield10K_LMK1_CPS250, payShield10K_LMK1_CPS2500, payShield10K_LMK2_CPS60, payShield10K_LMK2_CPS250, payShield10K_LMK2_CPS2500"
```
Comment (bmoore-msft) [inline quickstarts/microsoft.hardwaresecuritymodules/payment-hsm-create/azuredeploy.json:29]: use allowedValues to limit the list

**Review by bmoore-msft — CHANGES_REQUESTED** [review]

**Comment by azure-quickstarts** [comment]
@msmbaldwin - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

File: quickstarts/microsoft.hardwaresecuritymodules/payment-hsm-create/azuredeploy.json (hunk 3 above)
Comment (msmbaldwin) [reply quickstarts/microsoft.hardwaresecuritymodules/payment-hsm-create/azuredeploy.json]: Done.

File: quickstarts/microsoft.hardwaresecuritymodules/payment-hsm-create/azuredeploy.parameters.json (hunk 1 above)
Comment (msmbaldwin) [reply quickstarts/microsoft.hardwaresecuritymodules/payment-hsm-create/azuredeploy.parameters.json:4]: Added defaults.

File: quickstarts/microsoft.hardwaresecuritymodules/payment-hsm-create/azuredeploy.json (hunk 2 above)
Comment (msmbaldwin) [reply quickstarts/microsoft.hardwaresecuritymodules/payment-hsm-create/azuredeploy.json:132]: enableDdosProtection? I only see it once.

Comment (bmoore-msft) [reply quickstarts/microsoft.hardwaresecuritymodules/payment-hsm-create/azuredeploy.json:132]: The hsmSubnet is defined on the vnet resource as an array element and also as a standalone child resource...

**Comment by azure-quickstarts** [comment]
@msmbaldwin - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

**Comment by azure-quickstarts** [comment]
@msmbaldwin - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

**Pull request merged**